File: MezonQuiz/src/WebApp/Application/Dtos/UserDto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApp.Application.Dtos
{
    public class CreateUserRequestDto
    {
        public string? Email { get; set; }
        [Required]
        public string Username { get; set; } = null!;
        [Required]
        public string Password { get; set; } = null!;
        public string? DisplayName { get; set; }
        public string? AvatarUrl { get; set; }
    }

    public class UpdateUserRequestDto
    {
        public string? Email { get; set; }
        public string? DisplayName { get; set; }
        public string? AvatarUrl { get; set; }
        public bool IsActive { get; set; }
    }

    public class UserDto
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string? MezonUserId { get; set; }

        public string? Email { get; set; }

        [Required]
        public string Username { get; set; } = null!;

        public string? DisplayName { get; set; }

        public string? AvatarUrl { get; set; }

        public bool IsActive { get; set; }

        public DateTime? LastLoginAt { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
    }
}

File: MezonQuizFE/src/Interface/User.dto.ts
export interface CreateUserRequest {
  email?: string;
  username: string;
  password: string;
  displayName?: string;
  avatarUrl?: string;
}

export interface UpdateUserRequest {
  email?: string;
  displayName?: string;
  avatarUrl?: string;
  isActive: boolean;
}

export interface UserResponse {
  id: string;
  mezonUserId?: string;
  email?: string;
  username: string;
  displayName?: string;
  avatarUrl?: string;
  isActive: boolean;
  lastLoginAt?: string;
  createdAt: string;
  updatedAt: string;
}